namesys/namesys.go

```diff
@@ -1,59 +1,95 @@
 package namesys
 
 import (
+	"strings"
+
 	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
 	ci "github.com/ipfs/go-ipfs/p2p/crypto"
 	path "github.com/ipfs/go-ipfs/path"
 	routing "github.com/ipfs/go-ipfs/routing"
 )
 
-// ipnsNameSystem implements IPNS naming.
+// ipfs NameSystem implements generic IPFS naming.
```

Comment:
ah yes. this is a bit of a tricky naming thing because-- yeah, ipns means mostly pki based ptrs, and "ipfs naming" could be defined as having many name resolvers (ipns + dns + ...). So far, it's been most useful to describe it to people as clearly isolating out `/ipfs/...` to be immutable _content-addressed_ stuff, and `/ipns/...` to be mutable name-resolved stuff. (agree that it's a bit unclean, but am not sure the naming switch helps. an alternative semantic construction may be to say that "ipns supports sfs-style pki based names, and other resolvers (dns + ...) for doing things like pretty names and so on."